doc/userguide/rules/rule-types.rst

```diff
@@ -0,0 +1,1146 @@
+                "fast_pattern": false,
+                "relative_next": true
+              }
+            },
+            {
+              "name": "byte_extract"
+            },
+            {
+              "name": "byte_test",
+              "byte_test": {
+                "nbytes": 4,
+                "offset": 4,
+                "base": "unset",
+                "flags": [
+                  "little_endian"
+                ]
+              }
+            }
+          ]
+        },
+        {
+          "name": "dce_stub_data",
+          "direction": "toserver",
+          "is_mpm": true,
+          "app_proto": "smb",
+          "progress": 0,
+          "matches": [
+            {
+              "name": "content",
+              "content": {
+                "pattern": "abc",
+                "length": 3,
+                "nocase": false,
+                "negated": false,
+                "starts_with": false,
+                "ends_with": false,
+                "is_mpm": true,
+                "no_double_inspect": false,
+                "fast_pattern": false,
+                "relative_next": true
+              }
+            },
+            {
+              "name": "byte_extract"
+            },
+            {
+              "name": "byte_test",
+              "byte_test": {
+                "nbytes": 4,
+                "offset": 4,
+                "base": "unset",
+                "flags": [
+                  "little_endian"
+                ]
+              }
+            }
+          ]
+        }
+      ],
+      "lists": {},
+      "mpm": {
+        "buffer": "dce_stub_data",
+        "pattern": "abc",
+        "length": 3,
+        "nocase": false,
+        "negated": false,
+        "starts_with": false,
+        "ends_with": false,
+        "is_mpm": true,
+        "no_double_inspect": false,
+        "fast_pattern": false,
+        "relative_next": true
+      }
+    }
+
+.. _detailed-flowcharts-sig-type:
+
+Detailed Flowcharts
+-------------------
+
+A look into the illustrated overall representation of functions or paths that
+determine signature types.
+
+.. _flowchart-ip-only-sig-type:
+
+IP Only and IP Only with negated addresses
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``ip_only`` and ``like_ip_only`` flows.
+
+.. image:: rule-types/IP-Only.png
+    :align: center
+    :alt: A flowchart representing the SignatureIsIPOnly function.
+
+.. _flowchart-pd-only-sig-type:
+
+Protocol Detection Only
+^^^^^^^^^^^^^^^^^^^^^^^
+
+``pd_only`` flow.
+
+.. image:: rule-types/PD-only.png
+    :align: center
+    :width: 400
+    :alt: A flowchart representing the SignatureIsPDOnly function.
+
+.. _flowchart-app-layer-packet-app-tx-stream-sig-types:
+
+Application Layer Protocol, Transaction, Packet, Stream and Stream-Packet rules
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``app_layer``, ``app_tx``, ``pkt``, ``stream`` and ``stream-pkt`` flows.
+
```

Comment:
require_packet and require_stream can be seen as flags "need_packet" and "need_stream" in the engine analysis output